backend/typescript/services/implementations/campService.ts

```diff
@@ -852,6 +850,111 @@ class CampService implements ICampService {
     }
   }
 
+  async deleteCampSessionsByIds(
+    campId: string,
+    campSessionIds: Array<string>,
+  ): Promise<void> {
+    try {
+      const campSessions: Array<CampSession> | null = await MgCampSession.find({
+        _id: {
+          $in: campSessionIds,
+        },
+        camp: {
+          _id: campId,
+        },
+      });
+
+      if (
+        campSessions == null ||
+        campSessions.length !== campSessionIds.length
+      ) {
+        throw new Error(
+          `Could not find all camp sessions with IDs in [${campSessionIds}] and belonging to camp with ID ${campId}.`,
+        );
+      }
+
+      const camp: Camp | null = await MgCamp.findById(campId);
+      if (!camp) {
+        throw new Error(`Camp with campId ${campId} not found.`);
+      }
+
+      // This is an edge case, where the camp session has a campId but the the session does not appear on that camp
+      const currentCampSessionsSet = new Set(
+        camp.campSessions.map((sessionObjectId) => sessionObjectId.toString()),
+      );
+
+      campSessionIds.forEach((sessionId) => {
+        if (!currentCampSessionsSet.has(sessionId)) {
+          throw new Error(
+            `Not all camp sessions with IDs in [${campSessionIds}] were found in camp with ID ${campId}.`,
+          );
+        }
+      });
+
+      const oldCampSessions = camp.campSessions; // clone the full array of campers for rollback
+
+      try {
+        // delete camper IDs from the array of campers in the camp session
```

Comment:
Only campSessions seem to be touched in this block of code

Reply:
Yeah it's a good callout. Will revisit - some inclination to use another service to do this work, but I think that would violate the idea of single transaction?